### PR TITLE
Fix/login with provider

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?
- Fixes login with google provider changing next-auth prisma adapter to a replica that can use our DB.
- Added new models Account, Session for future migration of DB to use next-auth prisma adapter.
- Manually deletes old cookie on logout page SSR.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Sign up with magic link
- [ ] Sign in with email/password credentials
- [ ] Sign up with email/password credentials
- [ ] Sign up with google provider
- [ ] Sign in with google provider
- [ ] Sign in with SAML credentials
- [ ] Sign in with any account with non https cookies(No __Secure-), navigate to 'auth/logout' it should no remove your current cookie session (This didn't happened before)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 1]
```diff
@@ -182,9 +183,10 @@ if (true) {
     })
   );
 }
-
+const calcomAdapter = CalComAdapter(prisma);
 export default NextAuth({
-  adapter: PrismaAdapter(prisma),
+  // @ts-ignore
```
Comment (alannnc) [inline apps/web/pages/api/auth/[...nextauth].tsx:188]: Required for the moment till we fully migrate our DB to use next-auth-prisma-adapter.

**Comment by emrysal** [comment]
After this migration is ran, what are the remaining steps needed doing to complete migration?

**Comment by alannnc** [comment]
> After this migration is ran, what are the remaining steps needed doing to complete migration?
@emrysal 
- Wait till most users create their Account row for google oauth, or normalize DB with current user identityProviders fields to their respective Account row.
- Handle refresh token rotation on oauth providers as next-auth don't do that automatically. More info: https://next-auth.js.org/tutorials/refresh-token-rotation
- If we want we can switch from Jwt to Db Sessions, so this means removing model and respective code. https://next-auth.js.org/adapters/models/#session

File: apps/web/lib/auth/next-auth-custom-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import { IdentityProvider } from "@prisma/client";
+import type { Adapter } from "next-auth/adapters";
+
+import { Account, Prisma, PrismaClient, User, VerificationToken } from "@calcom/prisma/client";
+
+import { identityProviderNameMap } from "@lib/auth";
+
+/** @return { import("next-auth/adapters").Adapter } */
```
Comment (agustif) [inline apps/web/lib/auth/next-auth-custom-adapter.ts:8]: What does this do?

File: apps/web/pages/api/auth/[...nextauth].tsx (hunk 1 above)
Comment (agustif) [reply apps/web/pages/api/auth/[...nextauth].tsx:188]: Nice work on the custom calcomAdapter 👍🏻

File: apps/web/pages/auth/logout.tsx [hunk 3]
```diff
@@ -51,6 +52,11 @@ export default function Logout(props: Props) {
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
   const ssr = await ssrInit(context);
+  // Deleting old cookie manually, remove this code after all existing cookies have expired
```
Comment (agustif) [inline apps/web/pages/auth/logout.tsx:55]: How can we know when all cookies are expired btw? What is max current expiryDate of cookies we can calculate it using that maybe

**Review by agustif — APPROVED** [review]
Code looks great @alannnc , hopefully with this we fix all our issues and can migrate to v4 fully!

Should we test in staging?

File: apps/web/lib/auth/next-auth-custom-adapter.ts (hunk 2 above)
Comment (alannnc) [reply apps/web/lib/auth/next-auth-custom-adapter.ts:8]: it's just JSDoc markup

**Comment by alannnc** [comment]
> Code looks great @alannnc , hopefully with this we fix all our issues and can migrate to v4 fully!
> 
> Should we test in staging?

@agustif Yeah, testing in staging it's a must. I find login a delicate entry point for our application

File: apps/web/pages/auth/logout.tsx (hunk 3 above)
Comment (alannnc) [reply apps/web/pages/auth/logout.tsx:55]: I suppose newly created ones, have the same time span configuration. We can take a look there.

File: apps/web/lib/auth/next-auth-custom-adapter.ts (hunk 2 above)
Comment (zomars) [reply apps/web/lib/auth/next-auth-custom-adapter.ts:8]: Basically is typescript without typescript

**Review by zomars — APPROVED** [review]

**Comment by zomars** [comment]
Won't merge until morning so I can monitor well.

**Pull request merged**